js/config.js

```diff
@@ -182,6 +187,159 @@ function saveUserConfig(fieldName, newValue, oldConfig) {
+            } catch (e) {
+                reject('can not parse config file data: ' + data + ', error: ' + err);
+                return;
+            }
+            resolve(userConfigData);
+        });
+    });
+}
+
+/**
+ * Method to overwrite user config on windows installer
+ * @param {String} perUserInstall - Is a flag to determine whether we are installing for per user
+ * @returns {Promise}
+ */
+function updateUserConfigWin(perUserInstall) {
+    return new Promise((resolve, reject) => {
+        const userConfigFile = path.join(app.getPath('userData'), configFileName);
+        // flag to determine whether per user installation
+        if (!perUserInstall) {
+            reject();
+            return;
+        }
+
+        // if user config file does't exists just copy global config file
+        if (!fs.existsSync(userConfigFile)) {
+            resolve(copyConfigWin());
+            return;
+        }
+
+        Promise.all([readGlobalConfig(), getUserConfigData(userConfigFile)])
+            .then((data) => {
+                resolve(updateUserConfig(data[0], data[1]));
+            })
+            .catch((err) => {
+                reject(err);
+            });
+    });
+}
+
+/**
+ * Method to overwrite user config on mac installer
+ * @param {String} globalConfigPath - The global config path from installer
+ * @returns {Promise}
+ */
+function updateUserConfigMac(globalConfigPath) {
+    return new Promise((resolve, reject) => {
+        const userConfigFile = path.join(dirs.userConfig(), configFileName);
+
+        // if user config file does't exists just copy global config file
+        if (!fs.existsSync(userConfigFile)) {
+            resolve(copyConfigMac(globalConfigPath));
+            return;
+        }
+
+        Promise.all([readGlobalConfig(), getUserConfigData(userConfigFile)])
+            .then((data) => {
+                resolve(updateUserConfig(data[0], data[1]));
+            })
+            .catch((err) => {
+                reject(err);
+            })
+    });
+}
+
+/**
+ * Method to copy global config file to user config directory for Windows
+ * @returns {Promise}
+ */
+function copyConfigWin() {
+    return new Promise((resolve, reject) => {
+        const globalConfigFileName = path.join('config', configFileName);
+        const execPath = path.dirname(app.getPath('exe'));
+        const globalConfigPath = path.join(execPath, '', globalConfigFileName);
+        const userConfigPath = app.getPath('userData');
+
+        childProcess.exec(`echo D|xcopy /y /e /s /c "${globalConfigPath}" "${userConfigPath}"`, { timeout: 60000 }, (err) => {
+            if (err) {
+                reject(err);
+                return;
+            }
+            resolve();
+        });
+    });
+}
+
+/**
+ * Method which copies global config file to user config directory for mac
+ * @param {String} globalConfigPath - The global config path from installer
+ * @returns {Promise}
+ */
+function copyConfigMac(globalConfigPath) {
```

Comment:
why do we need to pass globalConfigPath here - shouldn't it be relative to exe like code in readGlobalConfig?